exercises/grep/example.go

```diff
@@ -0,0 +1,102 @@
+package grep
+
+import (
+	"bufio"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type grepOptions struct {
+	n_line_numbers      bool
+	l_file_only         bool
+	i_any_case          bool
+	v_invert            bool
+	x_match_entire_line bool
+	prefix_filenames    bool
+}
+
+// GrepFiles searches for pattern in the files given, using optional flags.
+func GrepFiles(pattern string, flags []string, files []string) []string {
```

Comment:
`Files` might be a better name since it's `grep.GrepFiles` now

Reply:
Files seems not exactly what the function "does".
maybe grep.Search ?